tests/test_main.py

```diff
@@ -1,186 +1,165 @@
 
 
 @pytest.fixture
-def created_projects(sample_project: Project, sample_project_2: Project) -> Generator[list[ProjectDetails]]:
-    from src.main import projects
+def mock_project_2() -> Generator[ProjectDetails]:
+    yield ProjectDetails(project_id=uuid.uuid4(), name="Test Project", description="Test Description")
 
-    for project in (sample_project, sample_project_2):
-        project_details = ProjectDetails(**project.model_dump())
-        projects.append(project_details)
-    yield projects
 
+@pytest.fixture
+def project_data() -> Generator[Project]:
+    yield Project(name="Test Project", description="Test Description")
 
-def test_create_project(client: TestClient, sample_project: Project) -> None:
-    response = client.post("/projects", json=sample_project.model_dump())
-    assert response.status_code == 201
-    data = response.json()
-    expected_data = {
-        "name": sample_project.name,
-        "description": sample_project.description,
-        "project_id": data["project_id"],
-    }
-    assert data == expected_data
 
+@pytest.fixture
+def client(override_get_db: MagicMock) -> Generator[TestClient]:
+    yield TestClient(app)
 
-def test_get_projects(client: TestClient, created_projects: list[ProjectDetails]) -> None:
+
+def test_get_projects(client: TestClient, mock_db: MagicMock, mock_project: models.Projects) -> None:
+    mock_db.execute.return_value.scalars.return_value.all.return_value = [mock_project]
     response = client.get("/projects")
     assert response.status_code == 200
-    data = response.json()
     expected_data = [
-        {"name": project.name, "description": project.description, "project_id": str(project.project_id)}
-        for project in created_projects
+        {"project_id": str(mock_project.id), "name": mock_project.name, "description": mock_project.description}
     ]
-    assert data == expected_data
-
-
-def test_get_project_details(client: TestClient, created_project: ProjectDetails) -> None:
-    response = client.get(f"/project/{created_project.project_id}/info")
-    assert response.status_code == 200
-    data = response.json()
+    assert response.json() == expected_data
+
+
+def test_create_project(
+    client: TestClient, mock_db: MagicMock, project_data: Project, mock_project: models.Projects
+) -> None:
+    mock_db.add = MagicMock()
+    mock_db.commit = MagicMock()
+    mock_db.refresh = MagicMock()
+    with patch("src.main.create_project_", return_value=mock_project) as mock_create:
+        response = client.post("/projects", json=project_data.model_dump())
+        project_id = response.json()["project_id"]
+        expected_data = {
+            "project_id": project_id,
+            "name": mock_project.name,
+            "description": mock_project.description,
+        }
+        assert response.status_code == 201
+        assert response.json() == expected_data
+        mock_create.assert_called_once()
```

Comment:
You don't need this part in the patch, the actual call with invoke is on the line above.
Also use a separate var for response json so you dont parse response to json multiple times. You do that on lines 91 and 98